doc/holochain_101/src/dpki.md

```diff
@@ -0,0 +1,54 @@
+
+Addressing these needs is the function of Public Key Infrastructure, and in our case, because we use the power of Holochain itself to do this, a Distributed Public Key Infrastructure: DPKI.
+
+## Requirements
+
+DPKI needs to fulfill at least the following design requirements:
+
+1. Provide a way to create new keys for nodes.
+2. Provide a way to revoke compromised keys and re-issue keys for a node.
+3. Provide a way to verify the provenance of keys by grouping them as originating from a single actor.
+4. Securely manage the private keys.
+5. Reliably distribute and make available information about public keys.
+
+In designing a solution for DPKI we recognize that this is a complex and difficult enough problem that any solution will need to evolve, and in fact there will be multiple solutions necessary for different contexts.  Thus, we have built into the Holochain conductor a simple interface for the fundamental needed functions, e.g. creating new keys when installing a DNA for the first time, that can then be implemented by specialized DPKI applications.  Furthermore we've implemented a reference implementation of a Holochain based DPKI application, which we call DeepKey.
+
+## DeepKey
+
+To deliver on the basics of Distributed Public Key Infrastructure, we need a way to generate keys of various types (revocation, identity, encryption, signing) from seeds, and we need to be able to generate such seeds from primary seeds, so that a human agent can create related "device agents" provably under their control.
+
+After studying a number of uses cases, including initial sign-up, key revocation, etc, the central insight we came to was the need to create a Hierarchical Deterministic Key generation system, based on a Primary Seed, from which additional seeds can be generated which then are in turn used to actually generate many key-pairs.  This allows us, by-convention, to use the first seed generated by the Primary seed as the seed for revocation keys, and subsequent seeds as seeds for keys of separate Holochain devices that can be proven to be under the control of the holder of Primary Seed.
+
+DeepKey
+
+TODO: merge the various docs we developed to explain DeepKey here.
+- https://medium.com/holochain/part-2-holochain-holo-accounts-cryptographic-key-management-and-deepkey-bf32ee91af65
+- https://hackmd.io/UbfvwQdJRKaAHI9Xa7F3VA?view
```

Comment:
Are these hackmds editable by anyone who comes upon them?  We should check the permissions, since this will be published